api/src/neo4j/queries/gene.js

```diff
@@ -46,5 +47,94 @@ RETURN apoc.map.mergeList(COLLECT(value.data)) as gene
   return { ...gene, externalDbs: reformatExternalDbs(gene.externalDbs) };
 };
 
+const getGenesForHPA = async () => {
+  const statement = `
+MATCH (g:Gene:HumanGem)-[:V1_3_0]-(r:Reaction)-[:V1_3_0]-(s:Subsystem)-[:V1_3_0]-(ss:SubsystemState)
+USING JOIN ON r
+RETURN DISTINCT [g.id, ss.name, s.id]
+`;
+
+  return queryListResult(statement);
+};
+
+const BASE_URL = 'https://metabolicatlas.org';
+
+const getGeneDetailsForHPA = async ({ id }) => {
+  const statement = `
+MATCH (:Gene:HumanGem {id:'${id}'})-[:V1_3_0]-(r:Reaction)-[:V1_3_0]-(s:Subsystem)-[:V1_3_0]-(ss:SubsystemState)
+WITH DISTINCT s.id as sids, COLLECT(r.id) as rids, ss
+UNWIND sids as sid
+CALL apoc.cypher.run("
+  MATCH (:Subsystem {id: $sid})-[:V1_3_0]-(r:Reaction)-[:V1_3_0]-(cm:CompartmentalizedMetabolite)
+  USING JOIN ON r
+  WITH DISTINCT cm
+  MATCH (m:Metabolite)-[:V1_3_0]-(cm)-[:V1_3_0]-(c:Compartment)-[:V1_3_0]-(cs:CompartmentState)
+  USING JOIN ON c
+  RETURN DISTINCT { id: $sid, compartments: COLLECT(DISTINCT(cs.name)), model_metabolite_count: COUNT(DISTINCT(m.id)), compartment_metabolite_count: COUNT(DISTINCT(cm.id)) } as data
```

Comment:
It looks to me like there are a bit too many `DISTINCT` clauses here. For example, the `cm` should already be distinct on line 70, so maybe there is no need to run `DISTINCT` again before the count. Also, in terms of counts, I think graph nodes can be counted just fine, and might more efficient than counting the ids.

```suggestion
  RETURN DISTINCT { id: $sid, compartments: COLLECT(DISTINCT(cs.name)), model_metabolite_count: COUNT(DISTINCT(m)), compartment_metabolite_count: COUNT(DISTINCT(cm) } as data
```

Reply:
Absolutely, this was sloppy by me.